### DISCUSSION TITLE
AppGyver - GET RECORD | The server responded with a list for a single data record.

### DISCUSSION THREAD

**Discussion by ajcave71** [supabase/supabase#1143] [Questions]
Hi everybody. I am currently experimenting with Supabase and Appgyver. I am currently able to retrieve all records with the GET COLLECTION functionality using the defined API’s by Supabase.

While using the GET RECORD functionality with the API defined for get a record in supabase i am getting the following error-message: “The server responded with a list for a single data record.” while using the TEST functionality of GET RECORD.

Testing in the tool POSTMAN with the same API used in the GET RECORD i am receiving the expected record as result:

[
{
“id_beer”: 1,
“beer”: “Pareltje aan de Vliet”,
“beer_description”: “Deze American Pale Ale, met trots voor Voorschoten gebrouwen, is hoppig met tonen van citrus en is door z’n frisse afdronk een heerlijke doordrinker voor mooie zomerse dagen of een gezellige avond met je vrienden.”,
“beer_img_url”: “https://www.brouwerijcave.nl/1.jpg”,
“beer_style_ID”: 1,
“brewery_id”: 1
}
]

The config of GET RECORD is as follows:

Relative path: https://yocfawduejvcbfufblxf.supabase.co/rest/v1/Beers?id_beer=eq.1&select=*

Response key path: empty

Hope you have a solution…

Regards,
Antonio

**Comment by steve-chavez** [comment]
Hey @ajcave71,


I'm not familiar with AppGyver. But following the error message, perhaps it expects a json object instead of an array.

Try adding an `Accept: application/vnd.pgrst.object+json` header to your request.

**Reply by ajcave71** [reply]
Hi @steve-chavez,
Thanks for your prompt reply.
I added Accept: application/vnd.pgrst.object+json to the header which removes the [ ] (array) brackets from the response.

{
    "id_beer": 1,
    "beer": "Pareltje aan de Vliet",
    "beer_description": "Deze American Pale Ale, met trots voor Voorschoten gebrouwen, is hoppig met tonen van citrus en is door z'n frisse afdronk een heerlijke doordrinker voor mooie zomerse dagen of een gezellige avond met je vrienden.",
    "beer_img_url": "https://www.brouwerijcave.nl/1.jpg",
    "beer_style_ID": 1,
    "brewery_id": 1
}

Regards,
Antonio

**Reply by Chanda1459** [reply]
Hi @ajcave71 where you able to figure out a solution to your problem?  It's the exact same issue I have come across. Your input would be very much appreciated.

Regards,
Chanda
